dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SyntheticFileTree.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2024 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.dockstore.webservice.helpers;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Implements a memory-based FileTree, empty upon construction, to which
+ * files are added via the `addFile` method.  Useful to unit test code that
+ * uses FileTrees, without needing to create an actual tree of files (on
+ * GitHub, the local filesystem, within a resource file, etc) and then
+ * reference it via a compatible FileTree implementation.
+ */
+public class SyntheticFileTree implements FileTree {
+
+    private Map<String, String> pathToContent = new HashMap<>();
+    private static final String SLASH = "/";
```

Comment:
If I understand correctly, this is a file separator https://www.baeldung.com/java-file-vs-file-path-separator#1-get-the-file-separator

Reply:
Indeed, it's the Dockstore file path separator, which is always `/`.  Changed the name to be more indicative of the value.